src/main/java/dev/openfeature/javasdk/Value.java

```diff
@@ -0,0 +1,206 @@
+package dev.openfeature.javasdk;
+
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.List;
+
+import lombok.EqualsAndHashCode;
+import lombok.ToString;
+
+@ToString
+@EqualsAndHashCode
+@SuppressWarnings("PMD.BeanMembersShouldSerialize")
+public class Value {
+
+    private final Object innerObject;
+
+    public Value(Value value) {
+        this.innerObject = value.innerObject; 
+    }
+
+    public Value(Boolean value) {
+        this.innerObject = value; 
+    }
+
+    public Value(String value) {
+        this.innerObject = value; 
+    }
+
+    public Value(Integer value) {
+        this.innerObject = value; 
+    }
+
+    public Value(Double value) {
+        this.innerObject = value; 
+    }
+
+    public Value(Structure value) {
+        this.innerObject = value; 
+    }
+
+    public Value(List<Value> value) {
+        this.innerObject = value; 
+    }
+
+    public Value(ZonedDateTime value) {
+        this.innerObject = value != null 
+            ? value.format(DateTimeFormatter.ISO_ZONED_DATE_TIME) : null;
+    }
+
+    /** 
+     * Check if this Value represents a Boolean.
+     * 
+     * @return boolean
+     */
+    public boolean isBoolean() {
+        return this.innerObject instanceof Boolean;
+    }
+
+    /** 
+     * Check if this Value represents a String.
+     * 
+     * @return boolean
+     */
+    public boolean isString() {
+        return this.innerObject instanceof String;
+    }
+
+    /** 
+     * Check if this Value represents an Integer.
+     * 
+     * @return boolean
+     */
+    public boolean isInteger() {
+        return this.innerObject instanceof Integer;
+    }
+
+    /** 
+     * Check if this Value represents a Double.
+     * 
+     * @return boolean
+     */
+    public boolean isDouble() {
+        return this.innerObject instanceof Double;
+    }
+
+    /** 
+     * Check if this Value represents a Structure.
+     * 
+     * @return boolean
+     */
+    public boolean isStructure() {
+        return this.innerObject instanceof Structure;
+    }
+    
+    /** 
+     * Check if this Value represents a List.
+     * 
+     * @return boolean
+     */
+    public boolean isList() {
+        return this.innerObject instanceof List;
+    }
+
+    /** 
+     * Check if this Value represents a ZonedDateTime.
```

Comment:
Because it shares a storage format with string, would it be worthwhile to have a comment that dates are also strings? If you do care about dates, and you are using this in some kind of dispatch, then you would want to check isZonedDateTime before isString, or check both, or something to disambiguate.

Reply:
I'm wondering if it isn't better to simply store a datetime... Storing them as strings only adds this complication, as well as performance issues around parsing/error handling. It was stored as a string in the previous implementation and I didn't give it much thought when I translated it.

Reply:
I think storing it as a datetime, or removing it, are both good options.

Reply:
Now storing as ZonedDateTIme.